src/util/system.cpp

```diff
@@ -964,6 +964,31 @@ std::string ArgsManager::GetChainName() const
     return CBaseChainParams::MAIN;
 }
 
+void ArgsManager::LogArgs() const
```

Comment:
@ryanofsky You might have some thoughts on this?

Reply:
> @ryanofsky You might have some thoughts on this?

I don't object to this change, but it does seems a little unusual to write the config file to the debug log. Especially when you have to mask passwords and such.

For debugging purposes, my preferred alternative would be to have a `-printconfig` flag (similar to existing `-help` or `-version` flags) that would print the current settings in ini format. This could be more convenient for debugging because you'd be able to parse the config and get instant feedback without having to bring the daemon up and down. I was also thinking about this as a way of implementing A.J's suggestion from https://github.com/bitcoin/bitcoin/pull/15935#issuecomment-508156827 to help debug dynamic settings ("viewing/exporting the current dynamic settings [...], so you can paste them into your bitcoin.conf")